### PR TITLE
Adding a github-api image.

### PR DESCRIPTION
**Pull request by ybettan**
This image will contain the necessary tools for querying the Github API.

Signed-off-by: Yoni Bettan <yonibettan@gmail.com>

### CODE REVIEW

**Comment by ybettan** [comment]
I hope I did it correctly.

FYI @chaodaiG

File: images/github-api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2016 The Kubernetes Authors.
```
Comment (chaodaiG) [inline images/github-api/Dockerfile:1]: 2022, same as other added files

**Comment by chaodaiG** [comment]
> Once merge, will I just be able to consume it with `gcr.io/k8s-prow/github-api:latest` inside a Prow job?

that doesn't happen automatically, I think you'll need to add a prowjob to do so https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-test-infra.yaml

File: images/github-api/Dockerfile (hunk 1 above)
Comment (ybettan) [reply images/github-api/Dockerfile]: Fixed.

**Comment by ybettan** [comment]
/hold
I will add the job according to https://github.com/kubernetes/test-infra/pull/27390#issuecomment-1238345052

**Comment by ybettan** [comment]
Done.
/unhold

**Comment by chaodaiG** [comment]
/cc @BenTheElder 
as who has some opinions on this topic https://kubernetes.slack.com/archives/CDECRSC5U/p1662485270398029?thread_ts=1662380630.329009&cid=CDECRSC5U

**Comment by BenTheElder** [comment]
I'm not convinced test-infra should be hosting yet another image.

I also don't think the GitHub API is necessary for the tasks mentioned in the thread and I don't think we should be encouraging github API tokesn in presubmit.

**Comment by spiffxp** [comment]
>I'm not convinced test-infra should be hosting yet another image.

I agree. If there's a strong case to have something lightweight I could be convinced otherwise. But for purposes of trying something out I know `gcr.io/k8s-staging-infra-tools/k8s-infra` has all of these tools today; it's just a use-at-your-own-risk deal since it's intended solely for kubernetes/k8s.io jobs.

>I don't think we should be encouraging github API tokens in presubmit

This is the bigger issue. All presubmits currently run in the same cluster, same namespace.  If you were to provide a token, you incur the risk of untrusted code from some PR in some repo you don't know about using your token. We can lint this at a config level, but that's not a guarantee it doesn't happen.

Reading the slack thread it looks like you've decided to pursue an alternate approach so I'm going to close

/close

**Pull request closed without merging**